soc/nordic/nrf54h/ironside_se/include/nrf/ironside_se_boot_report.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SOC_NORDIC_NRF54H_IRONSIDE_SE_BOOT_REPORT_INCLUDE_NRF_IRONSIDE_SE_BOOT_REPORT_H_
+#define ZEPHYR_SOC_NORDIC_NRF54H_IRONSIDE_SE_BOOT_REPORT_INCLUDE_NRF_IRONSIDE_SE_BOOT_REPORT_H_
+
+#include <stdint.h>
+#include <stddef.h>
+
+#define IRONSIDE_SE_BOOT_REPORT_LOCAL_DOMAIN_CONTEXT_SIZE                                          \
+	CONFIG_SOC_NRF54H20_IRONSIDE_SE_BOOT_REPORT_LOCAL_DOMAIN_CONTEXT_SIZE
+#define IRONSIDE_SE_BOOT_REPORT_RANDOM_DATA_SIZE                                                   \
+	CONFIG_SOC_NRF54H20_IRONSIDE_SE_BOOT_REPORT_RANDOM_DATA_SIZE
+
+/** @brief UICR error description contained in the boot report. */
+struct ironside_se_boot_report_uicr_error {
+	/** The type of error. A value of 0 indicates no error */
+	uint32_t error_type;
+	/** Error descriptions specific to each type of UICR error */
+	union {
+		/** RFU */
+		struct {
+			uint32_t rfu[4];
+		} rfu;
+	} description;
+};
+
+/** @brief IRONside boot report. */
+struct ironside_se_boot_report {
+	/** Magic value used to identify valid boot report */
+	uint32_t magic;
+	/** Firmware version of IRONside SE. 8bit MAJOR.MINOR.PATCH.TWEAK */
+	uint32_t ironside_se_version_int;
+	/** Human readable extraversion of IRONside SE */
+	char ironside_se_extraversion[12];
+	/** Firmware version of IRONside SE recovery firmware. 8bit MAJOR.MINOR.PATCH.TWEAK */
+	uint32_t ironside_se_recovery_version_int;
+	/** Human readable extraversion of IRONside SE */
+	char ironside_se_recovery_extraversion[12];
+	/** Copy of SICR.UROT.UPDATE.STATUS.*/
+	uint32_t ironside_update_status;
+	/** See @ref ironside_se_boot_report_uicr_error */
+	struct ironside_se_boot_report_uicr_error uicr_error_description;
+	/** Data passed from booting local domain to local domain being booted */
+	uint8_t local_domain_context[IRONSIDE_SE_BOOT_REPORT_LOCAL_DOMAIN_CONTEXT_SIZE];
```

Comment:
I see that it's optional for this to be populated.

So I assume we need a local_domain_context_len to communicate how many bytes were written?

(0 when cpusec boots cpuapp for instance)

Reply:
Alternatively we could make this mandatory.

Reply:
IMO there is a contract between the image being booted and the image that boots in this situation. For instance, for sec -> app we know that there is no local domain context.

For app -> rad there would be a struct that would communicate within in whether or not there is data of interest within the local domain context.